pulsar/producer.go

```diff
@@ -199,3 +204,15 @@ type Producer interface {
 	// of errors, pending writes will not be retried.
 	Close()
 }
+
+// GetHashingFunction return the corresponding hashing function for the hashing scheme
+func GetHashingFunction(s HashingScheme) func(string) uint32 {
```

Comment:
Why do we need to expose this publicly? An application is free to use a custom message router function, at that point it can use any hashing function.

Reply:
We would like to mimic what the default router does including the hash function. The only difference would be to use a snapshotted numPartitions instead of a latest one that is from `TopicMetadata`: https://github.com/apache/pulsar-client-go/blob/6ab17dcca28db389ab91a4a2ea6e5ae990d613da/pulsar/producer_impl.go#L103